File: c++/Processor.cpp
#include <iostream>
#include <string>

using namespace std;

class Processor
{
    private:
        /* atribut private */
        string name;
        int price;
    public:
        Processor(){
        }

        Processor(string n, int p){
            name = n;
            price = p;
        }

        //setter
        void setName(string n){
            name = n;
        }
        void setProcessorPrice(int p){
            price = p;
        }

        //getter
        string getName(){
            return name;
        }
        int getProcessorPrice(){
            return price;
        }

        //print atribute
        void showProcessor(){
            cout << "Processor Name  : " << name << endl;
            cout << "Processor Price : Rp. " << price << endl;
        }

        ~Processor(){}
};

File: java/Processor.java
public class Processor {
    
    private String name;
    private int price;
    
    // constructor
    Processor(){
    }

    Processor(String n, int p){
        name = n;
        price = p;
    }

    // Getter and Setter 
    public String getName() {
        return name;
    }
    public int getPrice() {
        return price;
    }
    public void setPrice(int price) {
        this.price = price;
    }
    public void setName(String name) {
        this.name = name;
    }

    // Prosedur untuk menampilkan atribut kelas Processor
    public void showProcessor() {
        System.out.println("Processor Name  : " + name);
        System.out.println("Processor Price : Rp. " + price);
    }
}
